lib/scala/kg/src/main/scala/io/github/tetherlessworld/mcsapps/lib/kg/stores/postgres/AbstractPostgresKgStore.scala

```diff
@@ -18,16 +19,34 @@ abstract class AbstractPostgresKgStore(protected val databaseConfigProvider: Pos
   protected lazy val edges = TableQuery[EdgeTable]
   protected lazy val edgeLabels = TableQuery[EdgeLabelTable]
   protected lazy val edgeSources = TableQuery[EdgeSourceTable]
-  protected lazy val nodes = TableQuery[NodeTable]
-  protected lazy val nodeLabels = TableQuery[NodeLabelTable]
+  protected object nodes extends TableQuery(new NodeTable(_)) {
+    def withNodeLabels = for {
+      ((node, _), nodeLabel) <- nodes
+        .join(nodeNodeLabels).on(_.id === _.nodeId)
+          .join(nodeLabels).on(_._2.nodeLabelLabel === _.label)
+    } yield (node, nodeLabel)
+
+    def withSources = for {
```

Comment:
Same thing here, I'm concerned that this is going to select the whole join table first before filtering it.